fix: #6380 遗留——task_processor 测试断言级别错(WARN≠ERROR)必失败 + list_paper_accounts 漏 HTTPException 透传 500→400

## 背景

#6380（fix #5479）合并后发现两处遗留问题（合并 commit `544c851c`）。审查 comment 见 #6380。

## 问题 1（P1）：新增单元测试 `test_on_progress_logs_on_stats_failure` 在 master 上必失败

`tests/unit/workers/test_task_processor_on_progress.py` 断言 `m_glog.ERROR.called`，但 `BacktestProcessor._on_progress`（`src/ginkgo/workers/backtest_worker/task_processor.py:293`）统计失败分支用的是 `GLOG.WARN`（#6031 引入，注释「对齐 #6031：不再静默吞异常」）。#6380 未改动 `task_processor.py`，测试与实现不匹配。

实测（master HEAD `544c851c`）：
```
tests/unit/workers/test_task_processor_on_progress.py::test_on_progress_logs_on_stats_failure
>   assert m_glog.ERROR.called, "统计失败应记 GLOG.ERROR 而非静默 pass"
E   AssertionError: 统计失败应记 GLOG.ERROR 而非静默 pass
```

注：CI 只跑 Smoke + py_compile，不跑 `tests/unit/`，故此失败未被 CI 拦截——任何本地跑 `pytest tests/unit/workers/` 都会红。

### 复现
```bash
/home/kaoru/.ginkgo/.venv/bin/python -m pytest tests/unit/workers/test_task_processor_on_progress.py -o addopts=""
```

### 修法（二选一）
- **推荐**：断言改为 `m_glog.WARN.called`，对齐 #6031 现状。统计聚合失败非致命、主流程继续上报，WARN 级别合理。
- 或把实现 `GLOG.WARN → GLOG.ERROR`（需评估与 #6031 日志级别约定是否冲突；倾向不改实现）。

## 问题 2（P2）：`list_paper_accounts` 漏 `except HTTPException: raise`，500 被吞成 400

#6380 给 `get_paper_account` / `get_paper_positions` / `get_paper_orders` 三端点加了 `except HTTPException: raise` 透传，但 `list_paper_accounts`（`api/api/trading.py:194`）同样在循环里调用 `_query_positions(p.uuid)`（`:211`），其异常链 `except BusinessError`（:232）→ `except Exception`（:234）唯独漏了 HTTPException 透传。

后果：helper 改 raise 之后，`list_paper_accounts` DB 故障从「静默返 200（值 0）」变成「`BusinessError(400)`」——非 AC1 要求的 500，且循环中任一账户持仓查询失败会让整个列表端点 400。

### 修法
在 `list_paper_accounts` 的 `except BusinessError: raise` 之前或之后加 `except HTTPException: raise`，与其他三端点对齐。

## 验收
- [ ] `test_on_progress_logs_on_stats_failure` 通过（断言与实现对齐）
- [ ] `list_paper_accounts` DB 故障返 500 非 400
- [ ] `pytest tests/unit/workers/test_task_processor_on_progress.py tests/api/test_trading_helpers.py` 全绿

> *This was generated by AI during triage.*

## Agent Brief

**Category:** bug
**Summary:** 修 #6380 两处遗留——①单元测试断言日志级别错（WARN≠ERROR）在 master 必失败；②`list_paper_accounts` 端点漏 `except HTTPException: raise`，DB 故障 500 被吞成 400。

**复现状态（已 triage 实测确认，origin/master HEAD `544c851c`）：**
- ①实跑 `pytest tests/unit/workers/test_task_processor_on_progress.py -o addopts=""` 失败：`AssertionError: 统计失败应记 GLOG.ERROR`，`m_glog.ERROR.called == False`。实现确实用 `GLOG.WARN`。
- ②`list_paper_accounts` 的异常链是 `except BusinessError: raise` → `except Exception`，唯独缺 `except HTTPException: raise`；三个 sibling 端点（`get_paper_account`/`get_paper_positions`/`get_paper_orders`）#6380 已加。`_query_positions` 在 DB 故障时 `raise HTTPException(500)`，故本端点循环内任一账户查询失败→HTTPException 落入 `except Exception`→改写成 `BusinessError(400)`，且整个列表端点 400。

**Current behavior:**
1. 测试 `test_on_progress_logs_on_stats_failure` 断言 `GLOG.ERROR.called`，但 `BacktestProcessor._on_progress` 统计失败分支用 `GLOG.WARN`（#6031 约定：不再静默吞异常，但非致命故 WARN 级）。测试与实现不匹配，master 上红。
2. `list_paper_accounts` 缺 `except HTTPException: raise`，DB 故障返回 400 而非 AC1 要求的 500。

**Desired behavior:**
1. 测试断言对齐实现的实际级别（WARN），与 #6031 约定一致。**实现不改**——统计聚合失败非致命、主流程继续上报，WARN 语义正确。
2. `list_paper_accounts` 加 `except HTTPException: raise`，HTTPException 原样透传（500 不变），与三个 sibling 端点对齐。

**Key interfaces:**
- `BacktestProcessor._on_progress`（`workers/backtest_worker/task_processor.py`）——保持 `GLOG.WARN`，**不要**改 ERROR。
- 单元测试 `test_on_progress_logs_on_stats_failure`——断言改 `WARN.called`，docstring 同步。
- FastAPI 端点 `list_paper_accounts`（`api/api/trading.py`）——异常链加 `except HTTPException: raise`，照搬 `get_paper_account`/`get_paper_positions`/`get_paper_orders` 的写法。
- `_query_positions` helper——已 `raise HTTPException(500)`，**无需改动**。

**Acceptance criteria:**
- [ ] `pytest tests/unit/workers/test_task_processor_on_progress.py -o addopts=""` 通过（断言 WARN，对齐实现）
- [ ] `list_paper_accounts` 在 `_query_positions` raise HTTPException(500) 时原样透传（500 非 400）；在 `tests/api/test_trading_helpers.py` 加 `test_list_paper_accounts_endpoint_propagates_http_500`，照搬现有 `test_get_paper_positions_endpoint_propagates_http_500` 模板
- [ ] `pytest tests/unit/workers/test_task_processor_on_progress.py tests/api/test_trading_helpers.py -o addopts=""` 全绿
- [ ] `BacktestProcessor._on_progress` 日志级别保持 WARN 不变

**Out of scope:**
- 把实现 `GLOG.WARN` 改成 `ERROR`（reporter 明确建议不改实现，WARN 语义正确）
- 其他 paper-trading 端点（#6380 已修）
- 跨 API 的 HTTPException 处理全仓审计（独立议题）

**注：** CI 只跑 Smoke + py_compile 不跑 `tests/unit/`，故 ① 类回归不会被 CI 拦截；agent 须本地实跑验收。

--- a/api/api/trading.py
+++ b/api/api/trading.py
@@ -231,6 +231,8 @@ async def list_paper_accounts():
 
     except BusinessError:
         raise
+    except HTTPException:
+        raise
     except Exception as e:
         logger.error(f"Error listing paper accounts: {str(e)}")
         raise BusinessError(f"Error listing paper accounts: {str(e)}")

--- a/tests/api/test_trading_helpers.py
+++ b/tests/api/test_trading_helpers.py
@@ -121,3 +121,31 @@ def test_get_paper_account_endpoint_propagates_http_500(api_modules):
         with pytest.raises(HTTPException) as exc:
             asyncio.run(trading_mod.get_paper_account("acct-1"))
     assert exc.value.status_code == 500
+
+
+# #6495: #6380 给 get_paper_account/get_paper_positions/get_paper_orders 三端点加了
+# except HTTPException: raise，但 list_paper_accounts（循环内调 _query_positions）漏改，
+# DB 故障被 except Exception 吞成 BusinessError(400) 非 AC1 的 500。
+# 修法：异常链补 except HTTPException: raise，与三 sibling 对齐。
+def test_list_paper_accounts_endpoint_propagates_http_500(api_modules):
+    """AC1: list_paper_accounts 端点同型透传 HTTPException(500)（循环内 _query_positions raise）。"""
+    import asyncio
+    import api.trading as trading_mod
+    from ginkgo.data.services.base_service import ServiceResult
+
+    # list_paper_accounts 经 portfolio_service.get(mode=PAPER) 拿账户列表，循环内调 _query_positions
+    mock_p = MagicMock()
+    mock_p.uuid = "p1"
+    mock_p.initial_capital = 100000
+    mock_p.cash = 100000
+    mock_p.name = "paper-1"
+
+    svc = MagicMock()
+    svc.get.return_value = ServiceResult.success([mock_p])
+
+    with patch.object(trading_mod, "_get_portfolio_service", return_value=svc), \
+         patch.object(trading_mod, "_query_positions",
+                      side_effect=HTTPException(status_code=500, detail="DB down")):
+        with pytest.raises(HTTPException) as exc:
+            asyncio.run(trading_mod.list_paper_accounts())
+    assert exc.value.status_code == 500

--- a/tests/unit/workers/test_task_processor_on_progress.py
+++ b/tests/unit/workers/test_task_processor_on_progress.py
@@ -3,7 +3,7 @@ from unittest.mock import MagicMock, PropertyMock, patch
 
 
 def test_on_progress_logs_on_stats_failure():
-    """统计聚合异常 → GLOG.ERROR 被调用（非静默 pass），主流程仍上报进度。"""
+    """统计聚合异常 → GLOG.WARN 被调用（非静默 pass），主流程仍上报进度。"""
     from ginkgo.workers.backtest_worker import task_processor as tp_mod
     from ginkgo.workers.backtest_worker.task_processor import BacktestProcessor
 
@@ -30,5 +30,5 @@ def test_on_progress_logs_on_stats_failure():
     with patch.object(tp_mod, "GLOG") as m_glog:
         proc._on_progress(0.5, "2026-01-01")  # 不应抛（主流程继续）
 
-    assert m_glog.ERROR.called, "统计失败应记 GLOG.ERROR 而非静默 pass"
+    assert m_glog.WARN.called, "统计失败应记 GLOG.WARN 而非静默 pass"
     assert progress_tracker.report_progress.called, "主流程上报不应中断"

